### PR TITLE
refs-and-the-dom guide translation

### PR DESCRIPTION
**Pull request by wogsland**

### CODE REVIEW

File: content/docs/refs-and-the-dom.md [hunk 1]
```diff
@@ -11,33 +11,33 @@ redirect_from:
 permalink: docs/refs-and-the-dom.html
 ---
 
-Refs provide a way to access DOM nodes or React elements created in the render method.
+Refs bieten eine Möglichkeit, auf DOM-Knotens oder React-Elemente zuzugreifen, die mit der Rendermethode erstellt wurden.
```
Comment (jakeherp) [inline content/docs/refs-and-the-dom.md]:
```suggestion
Refs bieten eine Möglichkeit, auf DOM-Knoten oder React-Elemente zuzugreifen, die mit der Rendermethode erstellt wurden.
```

File: content/docs/refs-and-the-dom.md [hunk 2]
```diff
@@ -11,33 +11,33 @@ redirect_from:
...
 
-In the typical React dataflow, [props](/docs/components-and-props.html) are the only way that parent components interact with their children. To modify a child, you re-render it with new props. However, there are a few cases where you need to imperatively modify a child outside of the typical dataflow. The child to be modified could be an instance of a React component, or it could be a DOM element. For both of these cases, React provides an escape hatch.
+Im typischen React-Datenfluss sind [Props](/docs/components-and-props.html) die einzige Möglichkeit, wie übergeordnete Komponenten mit ihren untergeordneten Komponenten interagieren. Um ein untergeordnetes Element zu ändern, rendere es mit neuen Props neu. Es gibt jedoch einige Fälle, in denen du ein untergeordnetes Element unbedingt außerhalb des typischen Datenflusses ändern musst. Das zu ändernde untergeordnete Element kann eine Instanz einer React-Komponente oder ein DOM-Element sein. In beiden Fällen bietet React eine Notluke.
```
Comment (jakeherp) [inline content/docs/refs-and-the-dom.md]:
```suggestion
Im normalen React-Datenfluss stellen [Props](/docs/components-and-props.html) die einzige Möglichkeit dar, um übergeordnete Komponenten mit ihren untergeordneten Komponenten interagieren zu lassen. Um ein untergeordnetes Element zu verändern, muss es mit neuen Props erneut gerendert werden. Es gibt jedoch einige Fälle, in denen du ein untergeordnetes Element unbedingt außerhalb des typischen Datenflusses ändern musst. Das zu ändernde untergeordnete Element kann eine Instanz einer React-Komponente oder ein DOM-Element sein. In beiden Fällen bietet React einen Ausweg.
```

File: content/docs/refs-and-the-dom.md [hunk 3]
```diff
@@ -11,33 +11,33 @@ redirect_from:
...
+* Integration in DOM-Bibliotheken von Drittanbietern.
 
-Avoid using refs for anything that can be done declaratively.
+Vermeide die Verwendung von Refs für alles, was deklarativ durchgeführt werden kann.
 
-For example, instead of exposing `open()` and `close()` methods on a `Dialog` component, pass an `isOpen` prop to it.
+Anstatt beispielsweise die Methoden `open ()` und `close ()` in einer `Dialog` -Komponente verfügbar zu machen, übergebe ihr eine` isOpen`-Prop.
 
-### Don't Overuse Refs {#dont-overuse-refs}
+### Überbeanspruche Refs nicht {#dont-overuse-refs}
```
Comment (jakeherp) [inline content/docs/refs-and-the-dom.md]:
```suggestion
### Verwende Refs nicht zu häufig {#dont-overuse-refs}
```

File: content/docs/refs-and-the-dom.md [hunk 4]
```diff
@@ -11,33 +11,33 @@ redirect_from:
...
 
-Your first inclination may be to use refs to "make things happen" in your app. If this is the case, take a moment and think more critically about where state should be owned in the component hierarchy. Often, it becomes clear that the proper place to "own" that state is at a higher level in the hierarchy. See the [Lifting State Up](/docs/lifting-state-up.html) guide for examples of this.
+Deine erste Neigung könnte sein, Refs zu verwenden, um Dinge in deier App "geschehen zu lassen". Wenn dies der Fall ist, nehme dich einen Moment Zeit und überlege genauer, wo sich der Status in der Komponentenhierarchie befinden sollte. Oft wird klar, dass der richtige Ort, um diesen Zustand zu "besitzen", auf einer höheren Ebene in der Hierarchie liegt. Beispiele hierfür findest du im Handbuch [State anheben](/docs/lifting-state-up.html).
```
Comment (jakeherp) [inline content/docs/refs-and-the-dom.md]:
```suggestion
Deine erste Intuition könnte sein, refs zu verwenden, um Dinge in deiner App "geschehen zu lassen". Wenn dies der Fall ist, nehme dir einen Moment Zeit und überlege genauer, wo sich der Status in der Komponentenhierarchie befinden sollte. Oft wird klar, dass der richtige Ort, um diesen Zustand zu "besitzen", auf einer höheren Ebene in der Hierarchie liegt. Beispiele hierfür findest du im Handbuch [State anheben](/docs/lifting-state-up.html).
```

Comment (jakeherp) [reply content/docs/refs-and-the-dom.md]: Not happy with `um diesen Zustand zu "besitzen"`, but couldn't come up with a better alternative for "besitzen".

File: content/docs/refs-and-the-dom.md [hunk 5]
```diff
@@ -189,25 +190,25 @@ function CustomTextInput(props) {
...
 
-When possible, we advise against exposing DOM nodes, but it can be a useful escape hatch. Note that this approach requires you to add some code to the child component. If you have absolutely no control over the child component implementation, your last option is to use [`findDOMNode()`](/docs/react-dom.html#finddomnode), but it is discouraged and deprecated in [`StrictMode`](/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage).
+Wenn möglich, raten wir davon ab, DOM-Knoten freizulegen, dies kann jedoch eine nützliche Notluke sein. Beachte, dass du bei diesem Ansatz der untergeordneten Komponente Code hinzufügen musst. Wenn du absolut keine Kontrolle über die Implementierung der untergeordneten Komponente hast, kannst du als letzte Option [`findDOMNode ()`](/docs/react-dom.html#finddomnode) verwenden, dies wird jedoch in [`StrictMode`](/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage) nicht empfohlen und ist veraltet.
```
Comment (jakeherp) [inline content/docs/refs-and-the-dom.md]:
```suggestion
Wenn möglich, raten wir davon ab, DOM-Knoten freizulegen, dies kann jedoch ein nützlicher Ausweg sein. Beachte, dass du bei diesem Ansatz der untergeordneten Komponente Code hinzufügen musst. Wenn du absolut keine Kontrolle über die Implementierung der untergeordneten Komponente hast, kannst du als letzte Option [`findDOMNode ()`](/docs/react-dom.html#finddomnode) verwenden, dies wird jedoch in [`StrictMode`](/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage) nicht empfohlen und ist veraltet.
```

File: content/docs/refs-and-the-dom.md (hunk 4 above)
Comment (wogsland) [reply content/docs/refs-and-the-dom.md]: Why make Refs lowercase?

Comment (ph1p) [reply content/docs/refs-and-the-dom.md]: @jakeherp @wogsland I have changed it so that it can be merged.

**Comment by ph1p** [comment]
Thank you! 👍

**Pull request merged**